LinaTheExplorer/LinaTheExplorer/2. Hi bye device/RulesForDevice.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Hi_bye_device
+{
+    class RulesForDevice
+    {
+        public static List<Car> GenerateRules(List<Car> listOfCars)
```

Comment:
1. refactor the method

I see this method marks cars(drivers) based on applied check rules, and then the cars are filtered in `ShowSelectedCars` based on these marks. In fact, the marks are redundant. You can just filter the cars and return a list with only those which satisfy the rules. Then you can remove the field Car:SelectedDriver since will become unused

2. next, rename the method to better communicate its new behavior

After prev step is done, the method behavior is changed and now it selects cars from provided cars list, rather than creates some rules.
`GenerateRules` says the method produces some rules, ie creates new data, but now this name does not fit the new method's behavior. I propose you to rename the method to FilterCars